dockerfiles/theia/Dockerfile

```diff
@@ -82,6 +82,46 @@ RUN che:theia production
 # change permissions
 RUN find production -exec sh -c "chgrp 0 {}; chmod g+rwX {}" \; 2>log.txt
 
+
+###
+# Plugin Builder Image
+#
+FROM ${BUILD_ORGANIZATION}/${BUILD_PREFIX}-theia-dev:${BUILD_TAG} as plugins-builder
+WORKDIR ${HOME}
+
+# define in env variable GITHUB_TOKEN only if it is defined
+# else check if github rate limit is enough, else will abort requiring to set GITHUB_TOKEN value
+ARG GITHUB_TOKEN
+
+# Define che-theia branch to use for plugins
+ARG CHE_THEIA_PLUGIN_BRANCH=master
+
+# Check github limit
+RUN if [ ! -z "${GITHUB_TOKEN-}" ]; then \
+      export GITHUB_TOKEN=$GITHUB_TOKEN; \
+      echo "Setting GITHUB_TOKEN value as provided"; \
+    else \
+      export GITHUB_LIMIT=$(curl -s 'https://api.github.com/rate_limit' | jq '.rate .remaining'); \
+      echo "Current API rate limit https://api.github.com is ${GITHUB_LIMIT}"; \
+      if [ "${GITHUB_LIMIT}" -lt 10 ]; then \
+        printf "\033[0;31m\n\n\nRate limit on https://api.github.com is reached so in order to build this image, "; \
+        printf "the build argument GITHUB_TOKEN needs to be provided so build will not fail.\n\n\n\033[0m"; \
+        exit 1; \
+      else \
+        echo "GITHUB_TOKEN variable is not set but https://api.github.com rate limit has enough slots"; \
+      fi \
+    fi
+
+# Clone and build che-theia plugins
+RUN git clone --branch ${CHE_THEIA_PLUGIN_BRANCH}  --single-branch --depth 1 https://github.com/eclipse/che-theia ${HOME}/che-theia-source-code
+
+WORKDIR ${HOME}/che-theia-source-code/plugins/
+RUN yarn
+
+RUN mkdir -p ${HOME}/che-theia-plugins/ && \
+    find ${HOME}/che-theia-source-code/plugins/ -name "*.theia"  -exec sh -c "chgrp 0 {}; chmod g+rwX {}; cp {} ${HOME}/che-theia-plugins/" \; 2>log.txt
```

Comment:
for now it should contain a regexp to only select a subset of plug-ins

Reply:
I'm not sure also why we do chgrp/chmod there

Reply:
> for now it should contain a regexp to only select a subset of plug-ins

ok which ones do we keep ? @l0rd @evidolob @benoitf 
- [ ] containers-plugin
- [ ] factory-plugin
- [ ] ports-plugin
- [ ] ssh-plugin
- [ ] welcome-plugin

> I'm not sure also why we do chgrp/chmod there

not sure I thought we had a good reason as we do it for theia production package

Reply:
I mean, we do the chgrp/chmod for this folder later in production container so we don't need to do it in the build container

Reply:
ok i see :)

Reply:
@benoitf added the right filter and removed the permission change commands